workflows/l2vpn/shared/forms.py

```diff
@@ -12,14 +12,14 @@
 # limitations under the License.
 
 
-from orchestrator.forms.validators import Choice, choice_list
 from orchestrator.types import SubscriptionLifecycle
+from pydantic_forms.validators import Choice, choice_list
 
 from products.product_blocks.port import PortMode
-from workflows.shared import subscriptions_by_product_type_and_instance_value
+from workflows.shared import AllowedNumberOfL2vpnPorts, subscriptions_by_product_type_and_instance_value
 
 
-def ports_selector(number_of_ports: int) -> list:
+def ports_selector(number_of_ports: AllowedNumberOfL2vpnPorts) -> type[list[Choice]]:
```

Comment:
The annotation `number_of_ports: AllowedNumberOfL2vpnPorts` doesn't hurt, but it also doesn't validate anything. If that's what you expected, you could consider using [`@validate_call`](https://docs.pydantic.dev/latest/concepts/validation_decorator/)

We haven't used this in the SURF orchestrator, it is a nice idea actually.

Reply:
I added the more specific type to inform the reader of the code, not sure if it makes sense to actually validate it here and risk a exception.